app/src/main/java/de/grobox/transportr/settings/SettingsFragment.kt

```diff
@@ -108,4 +110,24 @@ class SettingsFragment : PreferenceFragmentCompat() {
         }
     }
 
+    override fun onDisplayPreferenceDialog(preference: Preference?) {
+        when (preference) {
+            // show a material design 3 dialog instead of the default md2 preference dialog
```

Comment:
The rest of preferences looks pretty much like MD2 still.

Reply:
Yep, Material Design 3 unfortunately doesn't support preference styling, that's why we need to do that dialog workaround. If we wanted to really make the preferences MD3, we would need to write the preference fragments from scratch, which probably nobody has interest in doing...